### PR TITLE
Update code to remove custom table support if data is already migrated

### LINKED ISSUE #984: Custom table compatibility code
On version [0.10.0](https://github.com/laterpay/laterpay-wordpress-plugin/releases/tag/v0.10.0) we added WP table schema support so laterpay doesn't need custom tables. We kept backwards compatibility and are maintaining it since then. 

New installations uses WP schema by default.

We should now force a user to migrate if they haven't done already using admin notice(added in 0.10.0) and keep only WP schema support in code. 

We can decide on which version we want to do that if we want to remove custom table support  @ksmoore125 

Disadvantage of keeping custom table support is overhead of maintaing two database schema for same query.

This will allow us to have same build as VIP [build](https://vip-svn.wordpress.com/plugins/laterpay/).

Currently, we are removing [compatibility](https://github.com/laterpay/laterpay-wordpress-plugin/tree/master/laterpay/application/Compatibility) folder from VIP build.

### PR DESCRIPTION
**Pull request by thrijith**
Fixes #984 

Updated notice for migration.
screenshot 2018-11-28 at 1 04 08 am

Message After migration.
screenshot 2018-11-28 at 1 04 21 am

Notice to clean up migration data ( Remove custom table support. ) will be shown only if data has been migrated.
screenshot 2018-11-28 at 1 04 31 am

**Edit**
Updated behavior

migration-cleanup

### CODE REVIEW

**Comment by PatelUtkarsh** [comment]
Do we need separate notice to clean up data? @thrijith 

Maybe if user has already migrated we remove custom tables in WP upgrade. 

If user is first time clicking migration notice we remove custom tables after all table migration is done.

File: laterpay/application/Compatibility/Migrate.php [hunk 1]
```diff
@@ -312,30 +303,23 @@ public function migrate_terms_table( $offset ) {
...
-     * @return void
+     * @return bool
      */
-    public function drop_custom_tables( LaterPay_Core_Event $event ) {
+    public function drop_custom_tables() {
 
-        if ( ! is_user_logged_in() || ! current_user_can( 'manage_options' ) ) {
-            return;
-        }
+        if ( laterpay_is_migration_complete() ) {
```
Comment (PatelUtkarsh) [inline laterpay/application/Compatibility/Migrate.php]: Should continue checking `current_user_can( 'manage_options' )`?

Comment (thrijith) [reply laterpay/application/Compatibility/Migrate.php]: `drop_custom_tables` won't be called if the user doesn't have permission.

https://github.com/laterpay/laterpay-wordpress-plugin/pull/1050/files#diff-1de5cf8560a526c845948882499a3f25R94

Won't that be enough?

Comment (PatelUtkarsh) [reply laterpay/application/Compatibility/Migrate.php]: If in installation script we are checking permission then it's fine.

Comment (thrijith) [reply laterpay/application/Compatibility/Migrate.php]: The ajax request ( which will call `drop_custom_tables ` if migration is complete ) won't process further if a user doesn't have permission. So no issue.

**Pull request merged**